.github/workflows/on-merge-group-and-push-branch-main.yml

```diff
@@ -1,7 +1,9 @@
-name: On Merge Group Branch "main"
+name: On Merge Group And Push Branch "main"
 
 on:
   merge_group:
+    types: [checks_requested]
```

Comment:
> Although only the checks_requested activity type is supported, specifying the activity type will keep your workflow specific if more activity types are added in the future.

https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#merge_group